Build operator image from non-Linux local OS

Hi.

I would like to contribute to this awesome project, so as a first thing, I set out to build the project as adviced in the contributing guide.

I'm using minikube (v1.3.1) on a mac and after setting up the environment, I run
`make install-minikube`

The build is going through fine, but the operator fails:

$ kubectl get pod
NAME                                                        READY   STATUS   RESTARTS   AGE
camel-k-operator-5f8644979d-5wb6l   0/1           Error        6                 5m58s

$ kubectl logs pod/camel-k-operator-5f8644979d-5wb6l
standard_init_linux.go:211: exec user process caused "exec format error"

I tried reverting to latest release 1.0.0-M1 and then everything works just fine. Can you please advice?

Thx for the quick reply. It makes sense - and yes, I'm on a mac.

If I execute

`GOOS=linux make images`

the build fails though. I'm reading through the cross compilation section of the go manual and I see that I also have to provide GOARCH=amd64. The build still fails though
`env GOOS=linux GOARCH=amd64 make images`
give
$ env GOOS=linux GOARCH=amd64 make images
go build -ldflags "-X github.com/apache/camel-k/pkg/cmd/operator.GitCommit=6f0da195533b1b848388493a0529d11db54855ad" -gcflags=-trimpath=/Users/jacobborella/go -asmflags=-trimpath=/Users/jacobborella/go -o kamel ./cmd/kamel/*.go
go test -c -tags="integration knative" ./e2e/*.go
go test ./...
?   	github.com/apache/camel-k/cmd/kamel	[no test files]
?   	github.com/apache/camel-k/cmd/util/json-schema-gen	[no test files]
?   	github.com/apache/camel-k/deploy	[no test files]
?   	github.com/apache/camel-k/pkg/apis	[no test files]
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b718/v1alpha1.test: exec format error
FAIL	github.com/apache/camel-k/pkg/apis/camel/v1alpha1	0.007s
?   	github.com/apache/camel-k/pkg/apis/camel/v1alpha1/knative	[no test files]
?   	github.com/apache/camel-k/pkg/apis/knative08compat	[no test files]
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b722/builder.test: exec format error
FAIL	github.com/apache/camel-k/pkg/builder	0.010s
?   	github.com/apache/camel-k/pkg/builder/kaniko	[no test files]
?   	github.com/apache/camel-k/pkg/builder/s2i	[no test files]
?   	github.com/apache/camel-k/pkg/client	[no test files]
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b725/cmd.test: exec format error
FAIL	github.com/apache/camel-k/pkg/cmd	0.015s
?   	github.com/apache/camel-k/pkg/cmd/builder	[no test files]
?   	github.com/apache/camel-k/pkg/cmd/operator	[no test files]
?   	github.com/apache/camel-k/pkg/controller	[no test files]
?   	github.com/apache/camel-k/pkg/controller/build	[no test files]
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b728/integration.test: exec format error
FAIL	github.com/apache/camel-k/pkg/controller/integration	0.009s
?   	github.com/apache/camel-k/pkg/controller/integrationkit	[no test files]
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b731/integrationplatform.test: exec format error
FAIL	github.com/apache/camel-k/pkg/controller/integrationplatform	0.026s
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b734/gzip.test: exec format error
FAIL	github.com/apache/camel-k/pkg/gzip	0.006s
?   	github.com/apache/camel-k/pkg/install	[no test files]
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b737/metadata.test: exec format error
FAIL	github.com/apache/camel-k/pkg/metadata	0.006s
?   	github.com/apache/camel-k/pkg/platform	[no test files]
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b740/trait.test: exec format error
FAIL	github.com/apache/camel-k/pkg/trait	0.005s
?   	github.com/apache/camel-k/pkg/util	[no test files]
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b743/camel.test: exec format error
FAIL	github.com/apache/camel-k/pkg/util/camel	0.011s
?   	github.com/apache/camel-k/pkg/util/cancellable	[no test files]
?   	github.com/apache/camel-k/pkg/util/defaults	[no test files]
?   	github.com/apache/camel-k/pkg/util/digest	[no test files]
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b746/envvar.test: exec format error
FAIL	github.com/apache/camel-k/pkg/util/envvar	0.007s
?   	github.com/apache/camel-k/pkg/util/finalizer	[no test files]
?   	github.com/apache/camel-k/pkg/util/indentedwriter	[no test files]
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b749/knative.test: exec format error
FAIL	github.com/apache/camel-k/pkg/util/knative	0.005s
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b752/kubernetes.test: exec format error
FAIL	github.com/apache/camel-k/pkg/util/kubernetes	0.005s
?   	github.com/apache/camel-k/pkg/util/kubernetes/customclient	[no test files]
?   	github.com/apache/camel-k/pkg/util/kubernetes/log	[no test files]
?   	github.com/apache/camel-k/pkg/util/log	[no test files]
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b755/maven.test: exec format error
FAIL	github.com/apache/camel-k/pkg/util/maven	0.012s
?   	github.com/apache/camel-k/pkg/util/minishift	[no test files]
?   	github.com/apache/camel-k/pkg/util/monitoring	[no test files]
?   	github.com/apache/camel-k/pkg/util/openshift	[no test files]
?   	github.com/apache/camel-k/pkg/util/source	[no test files]
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b758/sync.test: exec format error
FAIL	github.com/apache/camel-k/pkg/util/sync	0.008s
?   	github.com/apache/camel-k/pkg/util/tar	[no test files]
fork/exec /var/folders/p3/lt9l79r51cz7d43q98xp1kv00000gn/T/go-build838360257/b761/test.test: exec format error
FAIL	github.com/apache/camel-k/pkg/util/test	0.005s
?   	github.com/apache/camel-k/pkg/util/watch	[no test files]
FAIL
make: *** [test] Error 1

--- a/script/Makefile
+++ b/script/Makefile
@@ -155,13 +155,21 @@ lint:
 images: test
 	mkdir -p build/_maven_output
 	mkdir -p build/_output/bin
+ifneq ($(shell uname -s 2>/dev/null || echo Unknown),Linux)
+	GOOS=linux go build $(GOFLAGS) -o build/_output/bin/kamel ./cmd/kamel/*.go
+else
 	cp kamel build/_output/bin
+endif
 	operator-sdk build --image-builder docker $(IMAGE_NAME):$(VERSION)
 
 images-dev: test package-artifacts
 	mkdir -p build/_maven_output
 	mkdir -p build/_output/bin
+ifneq ($(shell uname -s 2>/dev/null || echo Unknown),Linux)
+	GOOS=linux go build $(GOFLAGS) -o build/_output/bin/kamel ./cmd/kamel/*.go
+else
 	cp kamel build/_output/bin
+endif
 	operator-sdk build --image-builder docker $(IMAGE_NAME):$(VERSION)
 
 images-push:

